src/chainrules.jl

```diff
@@ -38,9 +44,16 @@ end
 function _Ω_∂_xlog1py(x::Real, y::Real)
     log1py = log1p(y)
     z = x * log1py
-    Ω = iszero(x) && !isnan(y) ? zero(z) : z
+    yp1 = 1 + y
+    w = x / yp1
```

Comment:
It seems `yp1` is not used anywhere else:
```suggestion
    w = x / (1 + y)
```